src/SharpBrick.PoweredUp.BlueGigaBLE/BlueGigaBLEPoweredUpBluetoothAdapater.cs

```diff
@@ -0,0 +1,514 @@
+            //fires on reset/reboot
+            Bglib.BLEEventSystemBoot += new Bluegiga.BLE.Events.System.BootEventHandler(this.SystemBootEventHandler);
+            //fires on Connect-trial
+            Bglib.BLEEventConnectionStatus += new Bluegiga.BLE.Events.Connection.StatusEventHandler(ConnectionStatusHandler);
+            //fires on finding a service
+            Bglib.BLEEventATTClientGroupFound += new Bluegiga.BLE.Events.ATTClient.GroupFoundEventHandler(ServiceFoundHandler);
+            
+            //fires on finding an attribute/characteristic
+            Bglib.BLEEventATTClientFindInformationFound += new Bluegiga.BLE.Events.ATTClient.FindInformationFoundEventHandler(CharacteristicFoundHandler);
+            //fires on end of service-discovery or end of characteristic-discovery
+            Bglib.BLEEventATTClientProcedureCompleted += new Bluegiga.BLE.Events.ATTClient.ProcedureCompletedEventHandler(ServiceOrCharacteristicSearchCompletedHandler);
+            Byte[] cmd = Bglib.BLECommandSystemReset(0x00); //boot normal, not in Device-Firmware-Update-mode!
+        }
+        #endregion
+        #region Initialization and System-/Serial-handler
+        private void InitSerial(String portname, int baudrate)
+        {
+            this.SerialAPI = new System.IO.Ports.SerialPort();
+            SerialAPI.Handshake = System.IO.Ports.Handshake.RequestToSend;
+            SerialAPI.BaudRate = baudrate;
+            SerialAPI.PortName = portname;
+            SerialAPI.DataBits = 8;
+            SerialAPI.StopBits = System.IO.Ports.StopBits.One;
+            SerialAPI.Parity = System.IO.Ports.Parity.None;
+            //this interprets the serial data as BlueGiga-Packets; see DataReceivedHandler
+            SerialAPI.DataReceived += new System.IO.Ports.SerialDataReceivedEventHandler(DataReceivedHandler);
+            SerialAPI.Open();
+        }
+        public async void SystemBootEventHandler(object sender, Bluegiga.BLE.Events.System.BootEventArgs e)
+        {
+            //according to documentation this event is not fired when USB-connected???
+            String log = String.Format("SystemBootEvent:" + Environment.NewLine + "\tmajor={0}" + Environment.NewLine +"\tminor={1}" + Environment.NewLine +"\tpatch={2}" + Environment.NewLine +"\tbuild={3}" + Environment.NewLine +"\tll_version={4}" + Environment.NewLine +"\tprotocol_version={5}" + Environment.NewLine +"\thw={6}" + Environment.NewLine,
+                e.major,
+                e.minor,
+                e.patch,
+                e.build,
+                e.ll_version,
+                e.protocol_version,
+                e.hw
+                );
+            if (TraceDebug) _logger?.LogDebug(log);
+            BleStatus.ActualStatus = BlueGigaBLEStatus.STATE_STANDBY;
+        }
+
+        private void DataReceivedHandler(object sender, System.IO.Ports.SerialDataReceivedEventArgs e)
+        {
+            System.IO.Ports.SerialPort sp = (System.IO.Ports.SerialPort)sender;
```

Comment:
use file wide usings instead of fully qualifying a type in c# code.

Reply:
simplified whereever I (or Visual Studio) saw it :-)